Backport tool charge helpers

S-S-X opened the pull request:
Base branch 1.2.x is selected just for sane comparison.

This PR _**backports following features**_ from (atow) master branch to current 1.2.2 stable release:
* `technic.get_RE_charge(itemstack)`
	* Returns current charge level of tool.
* `technic.set_RE_charge(itemstack, charge)`
	* Sets tool charge level.
* `technic.use_RE_charge(itemstack, charge)`
	* Attempt to use charge and return `true`/`false` indicating success.
	* Always succeeds without checking charge level if creative is enabled.

Aliases  that are _**not in Technic Plus Beta**_ but lately added to original Technic, happily had same interface as above stuff:
* technic.get_charge 
* technic.set_charge

Tool definition:
* `<tooldef>.technic_get_charge(stack)`.
* `<tooldef>.technic_set_charge(stack, charge)`.

### Additions

* technic.use_charge alias for consistency, see #392

### But why?

Mostly because following PR shows that we're bit too far behind with 1.2.x and 2.x would be a fairly big leap forward, there's version specific code for three versions (and this change would allow dropping one of those):
* https://github.com/mt-historical/bweapons_modpack/pull/1

~~Not~~ **included** simple things that ~~could be easily~~ have been backported for compatibility would be at least following tool def functions (just copypasta from master docs):
* For tool charger mods it is recommended to use `<tooldef>.technic_get_charge(stack)` instead.
* For tool charger mods it is recommended to use `<tooldef>.technic_set_charge(stack, charge)` instead.

## This probably should not be merged but instead
in case this gets approved it should make a new release branch, tag and cdb release.
Though in case someone argues this is a bug fix then yeah, 1.2.3 would be fine too.

S-S-X commented:
For test failures: integration tests probably shouldn't be touched and can't really be fixed, not worth it.
Pending luacheck... well that has not been even added yet (at this point it is still `lint`, not `luacheck`), or more like not yet renamed.
Simple mineunit checks and 1.2 luacheck passes. Though neither will actually test this stuff as here it is just for mod compatibility.

OgelGames commented:
Sounds good :+1: I'd be fine with bumping the version to 1.3.x for this, it's much more than a bug fix.

S-S-X commented:
Added:
* `<tooldef>.technic_get_charge(stack)`.
* `<tooldef>.technic_set_charge(stack, charge)`.
* `technic.use_charge` alias.

Updated PR description to reflect these additions.

S-S-X commented:
https://github.com/mt-mods/technic/releases/tag/1.3.0

S-S-X commented:
I think this should be ready for release.

For review, as I imported test fixtures and other test stuff from master and updated workflows there many test related changes.
For clean view it is possible to select just "Backport tool charge helpers" commit, that one contains all the mod related changes and excludes test related updates.

S-S-X commented:
Tested and done, https://github.com/mt-mods/technic/releases/tag/1.3.0

Though CDB release seemed to be stuck importing.
https://content.luanti.org/packages/mt-mods/technic_plus/
https://content.luanti.org/tasks/9ab0ee85-25d7-4b01-ba53-5a501f6ba7f1/?r=/packages/mt-mods/technic_plus/
